backend/Services/Services/RestaurantService.cs

```diff
@@ -146,7 +147,7 @@ public void UpdateRestaurant(Restaurant restaurant)
 
             if (restaurantDB == null)
             {
-                throw new Exception("Invalid id.");
+                throw new EntityNotFoundException("Invalid id.");
```

Comment:
same here

Reply:
also I think Restaurant variable name should be `restaurant` just like everywhere else, not `restaurantDB`

Reply:
Then it conflicts with parameter name. We would still need two Restaurant variable names, so I think it's ok for now.